src/tests/test_parse.py

```diff
@@ -298,28 +299,62 @@ def complete_with_defaults(mapping):
 
 
 @pytest.mark.parametrize(
-    "line, expected",
+    "line, cmd_from, expected_error_message",
     [
         (
-            "some-argument",
-            {"line": ["some-argument"]},
+            "duckdb:// --alias test1 --alias test2",
+            "sql",
+            (
+                "Duplicate arguments in %sql. "
+                "Please use only one of each of the following: --alias"
+            ),
         ),
         (
-            "a b c",
-            {"line": ["a", "b", "c"]},
+            """histogram --table penguins.csv --column bill_length_mm
+--column body_mass_g""",
+            "sqlplot",
+            (
+                "Duplicate arguments in %sqlplot. "
+                "Please use only one of each of the following: --column"
+            ),
         ),
+        (
+            """bar --table penguins.csv --column bill_length_mm
+--show-numbers --show-numbers""",
+            "sqlplot",
+            (
+                "Duplicate arguments in %sqlplot. "
+                "Please use only one of each of the following: --show-numbers"
+            ),
+        ),
+    ],
+)
+def test_magic_args_raises_usageerror(
+    load_penguin, ip, line, cmd_from, expected_error_message
+):
+    sql_line = ip.magics_manager.lsmagic()["line"][cmd_from]
+
+    with pytest.raises(UsageError) as excinfo:
+        magic_args(sql_line, line, cmd_from)
+    assert expected_error_message in str(excinfo.value)
+
+
+@pytest.mark.parametrize(
+    "line, expected_out",
+    [
+        ("some-argument", {"line": ["some-argument"]}),
+        ("a b c", {"line": ["a", "b", "c"]}),
         (
             "a b c --file query.sql",
             {"line": ["a", "b", "c"], "file": "query.sql"},
         ),
     ],
 )
-def test_magic_args(ip, line, expected):
+def test_magic_args_does_not_raise_usageerror(ip, line, expected_out):
```

Comment:
No need of renaming this.

Reply:
generally we do not append any suffix like `does_not_raise` for the success cases. You can check other tests

Reply:
Renamed function [here](https://github.com/ploomber/jupysql/pull/908/files#diff-edd543175e806ac6654196270596c90240cc3a3cfdcba20d35d997207e4555f0R353)